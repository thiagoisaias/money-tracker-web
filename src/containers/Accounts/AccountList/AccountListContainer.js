import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import AccountList from "../../../components/Accounts/AccountList/AccountList";

import { fetchAccounts } from "../../../store/actions/accounts/accounts";

export class AccountListContainer extends Component {
  componentDidMount() {
    const {
      accessToken,
      client,
      expiry,
      tokenType,
      uid,
      userId,
      onFetchAccountList
    } = this.props;

    const authHeaders = { accessToken, client, expiry, tokenType, uid };

    onFetchAccountList(userId, authHeaders);
  }

  render() {
    const { accountList, error, isLoading } = this.props;
    const propsToPass = { accountList, error, isLoading };
    return <AccountList {...propsToPass} />;
  }
}

AccountListContainer.propTypes = {
  accountList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      initialBalance: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired,
  error: PropTypes.string,
  userId: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  onFetchAccountList: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    accessToken: state.auth.accessToken,
    client: state.auth.client,
    expiry: state.auth.expiry,
    uid: state.auth.uid,
    userId: state.auth.userId,
    isLoading: state.accounts.isLoading,
    error: state.accounts.error,
    accountList: state.accounts.accountList
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onFetchAccountList: (userId, authHeaders) => {
      dispatch(fetchAccounts(userId, authHeaders));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  AccountListContainer
);
