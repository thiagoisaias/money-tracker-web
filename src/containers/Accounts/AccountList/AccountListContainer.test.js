import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { AccountListContainer } from "./AccountListContainer";

describe("AccountListContainer", () => {
  const mockProps = {
    accountList: [
      { id: "1", initialBalance: "34534", name: "Lorem One" },
      { id: "2", initialBalance: "94834", name: "Lorem Two" }
    ],
    userId: "1",
    error: null,
    isLoading: false,
    onFetchAccountList: jest.fn()
  };
  const component = shallow(<AccountListContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
