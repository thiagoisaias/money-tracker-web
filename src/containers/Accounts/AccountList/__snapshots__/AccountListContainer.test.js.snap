// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountListContainer renders properly 1`] = `
<Component
  accountList={
    Array [
      Object {
        "id": "1",
        "initialBalance": "34534",
        "name": "Lorem One",
      },
      Object {
        "id": "2",
        "initialBalance": "94834",
        "name": "Lorem Two",
      },
    ]
  }
  error={null}
  isLoading={false}
/>
`;
